app/frontend/src/store/tasks.ts

```diff
@@ -0,0 +1,50 @@
+import { createEffect, createEvent, createStore } from 'effector';
+import { createTask, getTasks, updateTask } from 'services/tasks';
+import { Task } from 'types/entities';
+
+
+/**
+ * Tasks store
+ */
+export const $tasks = createStore<Task[]>([]);
+
+/**
+ * Task effects
+ */
+export const getTasksFx = createEffect(getTasks);
+export const createTaskFx = createEffect(createTask);
+export const updateTaskFx = createEffect(updateTask);
+
+/**
+ * Event to be called when tasks list state should be updated
+ */
+export const listUpdated = createEvent<Task[]>();
+
+/**
+ * Sets tasks llist
+ */
+$tasks.on(getTasksFx.done, (_, { result }) => result.tasks);
+
+/**
+ * Add newly added task to tasks list
+ */
+$tasks.on(createTaskFx.done, (state, { result }) => [result.task, ...state]);
+
+/**
+ * Updates task in tasks list
+ */
+$tasks.on(updateTaskFx.done, (state, { result }) => {
+  // Find index of the task
+  const index = state.findIndex(item => result && result.task && item._id === result.task._id);
```

Comment:
What if index is not found?

Also looks like `result && result.task` could be moved to the top of the function as 
```
if (!result || !result.task) {
  return;
}
```

Reply:
don't know why but there is typescript error if i add such return statement

![image](https://user-images.githubusercontent.com/31101125/152041605-f4deb225-b3e7-4faf-b993-99c4d37addf9.png)